tracer/src/Datadog.Trace/Iast/Propagation/DefaultInterpolatedStringHandlerModuleImpl.cs

```diff
@@ -0,0 +1,110 @@
+// <copyright file="DefaultInterpolatedStringHandlerModuleImpl.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#if NET6_0_OR_GREATER
+
+#nullable enable
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Text;
+using Datadog.Trace.VendoredMicrosoftCode.System.Runtime.InteropServices;
+
+namespace Datadog.Trace.Iast.Propagation;
+
+internal static class DefaultInterpolatedStringHandlerModuleImpl
+{
+    public static unsafe void Append(IntPtr target, string? value)
+    {
+        FullTaintIfAnyTainted(target, value);
+    }
+
+    public static unsafe void FullTaintIfAnyTainted(IntPtr target, string? input)
+    {
+        try
+        {
+            IastModule.OnExecutedPropagationTelemetry();
+
+            if (input is null)
+            {
+                return;
+            }
+
+            var iastContext = IastModule.GetIastContext();
+            if (iastContext is null)
+            {
+                return;
+            }
+
+            var taintedObjects = iastContext.GetTaintedObjects();
+            var tainted = PropagationModuleImpl.GetTainted(taintedObjects, target);
+            var targetIsTainted = tainted is not null;
+
+            if (!targetIsTainted && (tainted = GetTaintedWithRanges(taintedObjects, input)) is null)
+            {
+                return;
+            }
+
+            // The start and length here are not correct, but we don't currently have a way to get the correct values
+            var rangesResult = new[] { new Range(0, 0, tainted!.Ranges[0].Source) };
```

Comment:
I'd set the range to cover the whole strings, as we do in other cases, like Format or Replace

Reply:
Updated in 30dfe3ea54544e9cdf7d86a8520ad53d97f8de7e
I changed the length to the result length like in Format or Replace 😄